src/main/java/uk/co/bconline/ndelius/service/impl/UserServiceImpl.java

```diff
@@ -109,10 +113,14 @@ public UserServiceImpl(
 		val dbFuture = supplyAsync(() -> userEntityService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 		val ldapFuture = supplyAsync(() -> userEntryService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 
+		val roleFuture = supplyAsync(() -> userRoleService.getAllUsersWithRole(role).stream()
+				.map(user -> LdapUtils.getStringValue(user.getDn(), user.getDn().size() - 2).toLowerCase()) // username is 2nd-to-last part of distinguished name
+				.collect(toList()));
```

Comment:
Small thing, would it simplify things to move this `.map` and `.collect` into `userRoleService::getAllUsersWithRole`, to save us collecting the stream twice?  So this would become:
```suggestion
		val roleFuture = supplyAsync(() -> userRoleService.getAllUsersWithRole(role))
```

and the service method would become something like:
```
	public List<String> getAllUsersWithRole(String role)
	{
		if (role == null || role.isBlank()) return emptySet();

		return stream(roleAssociationRepository.findAll(query()
				.searchScope(SUBTREE)
				.base(usersBase)
				.where(OBJECTCLASS).is("NDRoleAssociation").and("cn").like(role)).spliterator(), true)
				.map(user -> LdapUtils.getStringValue(user.getDn(), user.getDn().size() - 2).toLowerCase()) // username is 2nd-to-last part of distinguished name
				.collect(toList()));
	}
```

I've approved the PR so happy for it to go in as is, up to you if you want to make this change.